sklearn/svm/_classes.py

```diff
@@ -1316,3 +1319,227 @@ def probA_(self):
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+            Set of samples, where n_samples is the number of samples and
+            n_features is the number of features.
+
+        sample_weight : array-like of shape (n_samples,), default=None
+            Per-sample weights. Rescale C per sample. Higher weights
+            force the classifier to put more emphasis on these points.
+
+        y : Ignored
+            not used, present for API consistency by convention.
+
+        Returns
+        -------
+        self : object
+
+        Notes
+        -----
+        If X is not a C-ordered contiguous array it is copied.
+
+        """
+        super().fit(X, np.ones(_num_samples(X)),
+                    sample_weight=sample_weight, **params)
+        self.offset_ = -self._intercept_
+        return self
+
+    def decision_function(self, X):
+        """Signed distance to the enveloping hypersphere.
+
+        Signed distance is positive for an inlier and negative for an outlier.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            The data matrix.
+
+        Returns
+        -------
+        dec : ndarray of shape (n_samples,)
+            Returns the decision function of the samples.
+        """
+        dec = self._decision_function(X).ravel()
+        return dec
+
+    def score_samples(self, X):
+        """Raw scoring function of the samples.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            The data matrix.
+
+        Returns
+        -------
+        score_samples : ndarray of shape (n_samples,)
+            Returns the (unshifted) scoring function of the samples.
+        """
+        score_samples = self.decision_function(X) + self.offset_
+        return score_samples
+
+    def predict(self, X):
+        """Perform classification on samples in X.
+
+        For a one-class model, +1 or -1 is returned.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix} of shape (n_samples, n_features) or \
+                (n_samples_test, n_samples_train)
+            For kernel="precomputed", the expected shape of X is
+            (n_samples_test, n_samples_train).
+
+        Returns
+        -------
+        y_pred : ndarray of shape (n_samples,)
+            Class labels for samples in X.
+        """
+        y = super().predict(X)
+        return np.asarray(y, dtype=np.intp)
+
+    @deprecated(
+        "The probA_ attribute is deprecated in version 0.23 and will be "
+        "removed in version 0.25.")
+    @property
+    def probA_(self):
+        return self._probA
+
+    @deprecated(
+        "The probB_ attribute is deprecated in version 0.23 and will be "
+        "removed in version 0.25.")
+    @property
+    def probB_(self):
+        return self._probB
```

Comment:
Since this is a new class, there is no point in introducing deprecated attributes as nobody can have existing code using them. Just delete those lines.